From 9b0a2648c073e96b72734f6bd3fb82e4e4d7be80 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Mon, 3 Aug 2026 09:19:20 +0700
Subject: [PATCH 8/8] Revert "PCI/PM: Disable device wakeups when halting or
 powering off system"

This reverts commit 231e280ef82d08fe46bc5fe4c8a40071464a1122.
The reverted commit breaks WoL on ethernet controllers.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 drivers/pci/pci-driver.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/pci/pci-driver.c b/drivers/pci/pci-driver.c
index c0d6ffe79..7297e7bcf 100644
--- a/drivers/pci/pci-driver.c
+++ b/drivers/pci/pci-driver.c
@@ -1195,9 +1195,6 @@ static int pci_pm_poweroff(struct device *dev)
 	struct pci_dev *pci_dev = to_pci_dev(dev);
 	const struct dev_pm_ops *pm = dev->driver ? dev->driver->pm : NULL;
 
-	if (device_may_wakeup(dev) && system_state == SYSTEM_POWER_OFF)
-		device_set_wakeup_enable(dev, false);
-
 	pci_dev->skip_bus_pm = false;
 
 	if (pci_has_legacy_pm_support(pci_dev))
-- 
2.55.0

